fileutil/tarball_compressor.go

```diff
@@ -32,44 +37,130 @@ func (c tarballCompressor) CompressSpecificFilesInDir(dir string, files []string
 
 	defer tarball.Close()
 
-	tarballPath := tarball.Name()
+	zw := gzip.NewWriter(tarball)
+	tw := tar.NewWriter(zw)
+
+	for _, file := range files {
+		err = c.fs.Walk(filepath.Join(dir, file), func(f string, fi os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if filepath.Base(f) == ".DS_Store" {
+				return nil
+			}
+
+			header, err := tar.FileInfoHeader(fi, f)
+			if err != nil {
+				return bosherr.WrapError(err, "Reading tar header")
+			}
+
+			relPath, err := filepath.Rel(dir, filepath.ToSlash(f))
+			if err != nil {
+				return bosherr.WrapError(err, "Resovling relative tar path")
+			}
+			header.Name = relPath
+
+			if err := tw.WriteHeader(header); err != nil {
+				return bosherr.WrapError(err, "Writing tar header")
+			}
+
+			if fi.Mode().IsRegular() {
+				data, err := os.Open(f)
```

Comment:
should this opened file be closed or is handled by the close of the tarball `defer tarball.Close()`?